cmd/os-reliable.go

```diff
@@ -83,17 +83,20 @@ func renameAll(srcFilePath, dstFilePath string) (err error) {
 	}
 
 	if err = reliableRename(srcFilePath, dstFilePath); err != nil {
-		if isSysErrNotDir(err) {
+		switch {
```

Comment:
Sorry, I think you misunderstood what I meant. I wanted the `renameAll` function to look like this

func renameAll(srcFilePath, dstFilePath string) (err error) {
	if srcFilePath == "" || dstFilePath == "" {
		return errInvalidArgument
	}

	if err = checkPathLength(srcFilePath); err != nil {
		return err
	}
	if err = checkPathLength(dstFilePath); err != nil {
		return err
	}

	if err = reliableRename(srcFilePath, dstFilePath); err != nil {
		switch {
		case isSysErrNotDir(err):
			return errFileAccessDenied
		case isSysErrPathNotFound(err):
			// This is a special case should be handled only for
			// windows, because windows API does not return "not a
			// directory" error message. Handle this specifically here.
			return errFileAccessDenied
		case isSysErrCrossDevice(err):
			return fmt.Errorf("%s (%s)->(%s)", errCrossDeviceLink, srcFilePath, dstFilePath)
		case os.IsNotExist(err):
			return errFileNotFound
		default:
			return err
		}
	}
	return nil
}
```

Reply:
Oh, now it's clear 😉